src/utils/options.py

```diff
@@ -0,0 +1,313 @@
+            raise ValueError(errmsg)
+
+        self.status = params[1]  # 1 operational, -1 offline
+        self.geo_lat = params[4]  # decimal degrees, S = negative
+        self.geo_long = params[5]  # decimal degrees, W = negative
+        self.altitude = params[6]  # metres
+        self.boresight = params[7]  # degrees from geographic north, CCW = negative.
+        self.boresight_shift = params[8]  # degrees from physical boresight. nominal 0.0 degrees
+        self.beam_sep = params[9]  # degrees, nominal 3.24 degrees
+        self.velocity_sign = params[10]  # +1.0 or -1.0
+        self.phase_sign = params[11]  # +1 indicates correct interferometry phase, -1 indicates 180
+        self.tdiff_a = params[12]  # us for channel A.
+        self.tdiff_b = params[13]  # us for channel B.
+        # interferometer offset from midpoint of main, metres [x, y, z] where x is along line of
+        # antennas, y is along array normal and z is altitude difference, in m.
+        self.intf_offset = [float(params[14]), float(params[15]), float(params[16])]
+        self.analog_rx_rise = params[17]  # us
+        self.analog_rx_attenuator = params[18]  # dB
+        self.analog_atten_stages = params[19]  # number of stages
+        self.max_range_gates = params[20]
+        self.max_beams = params[21]  # so a beam number always points in a certain direction
+
+    def parse_restrict(self):
+        # Read in restrict.dat
+        if not os.environ["BOREALISPATH"]:
+            raise ValueError("BOREALISPATH env variable not set")
+        if not os.environ['RADAR_ID']:
+            raise ValueError('RADAR_ID env variable not set')
+        path = f'{os.environ["BOREALISPATH"]}/config/' \
+               f'{os.environ["RADAR_ID"]}/' \
+               f'restrict.dat.{os.environ["RADAR_ID"]}'
+        try:
+            with open(path) as data:
+                restricted = data.readlines()
+        except IOError:
+            print(f'IOError on restrict.dat file at {path}')
+            raise
+
+        restricted[:] = [line for line in restricted if line[0] != "#"]  # remove comments
+        restricted[:] = [line for line in restricted if len(line.split()) != 0]  # remove blanks
+
+        for line in restricted:
+            splitup = line.split("=")
+            if len(splitup) == 2:
+                if splitup[0] == 'default' or splitup[0] == 'default ':
+                    self.default_freq = int(splitup[1])  # kHz
+                    restricted.remove(line)
+                    break
+        else:  # no break
+            raise Exception('No Default Frequency Found in Restrict.dat')
+
+        self.restricted_ranges = []
+        for line in restricted:
+            splitup = line.split()
+            if len(splitup) != 2:
+                raise Exception('Problem with Restricted Frequency: A Range Len != 2')
+            try:
+                splitup = [int(float(freq)) for freq in splitup]  # convert to ints
```

Comment:
```suggestion
                splitup = [int(freq) for freq in splitup]  # convert to ints
```

Reply:
The float is used to cast decimal point numbers to a format that can then be casted to an integer. Without `float()`, if a frequency was listed as 150000.0, int() would throw an error.